pkg/gofr/datasource/sql/sql.go

```diff
@@ -122,13 +132,30 @@ func retryConnection(database *DB) {
 }
 
 func getDBConfig(configs config.Config) *DBConfig {
+	// if the value of maxIdleConn is negative or 0, no idle connections are retained.
+	maxIdleConn, err := strconv.Atoi(configs.Get("DB_MAX_IDLE_CONNECTION"))
+	if err != nil {
+		// setting the max open connection as the default which is being provided by default package
+		maxIdleConn = 2
```

Comment:
maybe you could use

const defaultMaxIdleConn = 2, with the comment you provided.so here you could use this here


```suggestion
		maxIdleConn = defaultMaxIdleConn
```

Code would be clearer, and it would be clearer you are not using a magical or arbitrary value.

Same remark for max open variable

Reply:
You could also document this as the default value, because even with an explanation, the 2 would sound to be magical number

https://github.com/golang/go/blob/4f77a83589baa5a3038cc6e35615084dc7e5f0c1/src/database/sql/sql.go#L952